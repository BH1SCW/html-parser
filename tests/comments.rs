use html_parser::prelude::*;
use indoc::indoc;
use insta::assert_debug_snapshot;

#[test]
fn it_can_parse_document_with_just_one_comment() -> Result<()> {
    let markup = "<!-- hello !\"#/()= -->";
    let ast = Ast::parse(markup)?;
    assert_debug_snapshot!(ast);
    Ok(())
}
#[test]
fn it_can_parse_document_with_just_comments() -> Result<()> {
    let markup = "<!--x--><!--y--><!--z-->";
    let ast = Ast::parse(markup)?;
    assert_debug_snapshot!(ast);
    Ok(())
}
