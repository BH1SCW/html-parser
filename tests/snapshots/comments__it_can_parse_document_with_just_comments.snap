---
source: tests/comments.rs
expression: ast
---
Ast {
    tree_type: Empty,
    nodes: [],
}
